File: parser/src/main/java/io/ticofab/androidgpxparser/parser/domain/RoutePoint.java
package io.ticofab.androidgpxparser.parser.domain;

/**
 * A route point (rtept) element.
 */
public class RoutePoint extends Point {

    private RoutePoint(Builder builder) {
        super(builder);
    }

    public static class Builder extends Point.Builder {

        @Override
        public RoutePoint build() {
            return new RoutePoint(this);
        }
    }

}

File: parser/src/main/kotlin/io/ticofab/androidgpxparser/parser/domain/RoutePoint.kt
package io.ticofab.androidgpxparser.parser.domain

/**
 * A route point (rtept) element.
 */
class RoutePoint private constructor(builder: Builder) : Point(builder) {
    class Builder : Point.Builder() {
        override fun build() = RoutePoint(this)
    }
}